src/const-safety.md

```diff
@@ -0,0 +1,171 @@
+# Const safety
+
+The miri engine, which is used to execute code at compile time, can fail in
```

Comment:
miri or Miri? I have often been writing the latter, but not consistently...^^

Reply:
I've fairly consistently used the lower case form. :shrug: I have no opinion on it beyond "my fingers like not using more modifier keys than they need to".

Reply:
More importantly though, I feel the first sentence is not a great introduction here. I wrote this in the context of the const-eval repo, but this is different.

I'd start with something like "As part of the compilation process, the Rust compiler has to evaluate some of the code of a program. This is called CTFE (compile-time function evaluation). CTFE is used for: array lengths, const/static initializers, ...". Then say that CTFE can go wrong and just like at run-time, safe Rust protects programmers from some of the ways CTFE can fail. "Programmers only writing safe code do not have to care, but if you want to write unsafe code that is run by CTFE, then there are some extra concerns you need to be aware of."

Then the following text makes much more sense IMO.^^